decompiler/frontend/binaryninja/handlers/globals.py

```diff
@@ -16,107 +16,28 @@ def register(self):
         """Register the handler at its parent lifter."""
         self._lifter.HANDLERS.update({DataVariable: self.lift_global_variable})
 
-    def lift_global_variable(self, variable: DataVariable, **kwargs) -> Union[UnaryOperation, GlobalVariable]:
-        """Lift a global variable.
-        kwargs should contain 2 keys:
-        parent_addr: an address in int if this is a recursive pointer, None otherwise.
-        bv: BinaryNinja BinaryView object."""
-        bv = kwargs["bv"]
-        parent_addr = kwargs["parent_addr"]
-        addr = variable.address
+    def lift_global_variable(self, variable: DataVariable, view: BinaryView, 
+        parent: Optional[MediumLevelILInstruction] = None, **kwargs
+    ) -> Union[ImportedFunctionSymbol, Constant, UnaryOperation]:
+        """Lift global variables with basic types (pointer are possible)"""
+        if not variable.name and isinstance(variable.value, bytes): # will only catch const char[x] stuff; may be better checking variable.type for ArrayType
```

Comment:
Could we ellaborate on this comment? Should we have a follow-up issue here?

Reply:
Already noted and will probably be a follow up issue. 
Tho, I only had one case where bninja yielded an array other than char (and which was unnamed)..

Reply:
Actually nvm, after looking a bit closer into my nodes, the comment is outdated. 

The main problem is: Lifting of strings in `scanf` (which was done here at some point, but is now done at the end of `lift_constant_pointer`). 

The old codes instantly lifts the strings inside of the method:
```C
   ....
   scanf("%x", &x);
   ....
```
where the new code, does the same as BNinja.
```C
  void* data_addr = "\x25\x78\x00";
   ....
   scanf(data_addr, &x);
   ....
```

I thought, that BNinja would give this kind of strings as a `const char[]/char*`, but Bninja does that _sometimes_, but most of the times it gives a `void`.

The assumptions the old code did, were not really good, resulting in stuff like this:
```C
   "%x" = "str"
```
Therefore i removed it for now. 
Will defienlinty make a follow up and look into BNinjas API to maybe find a way to fix it.